### PR TITLE
Bump to 3.4.0, update dependencies

### PR DESCRIPTION
**Pull request by nicornk**
Remove python-editor dependency and replace with editor dependency

Needs https://github.com/conda-forge/staged-recipes/pull/28074 to be merged first.

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by nicornk** [comment]
This is super strange. @conda-forge/help-python 
Can you help here? It seems pip somehow uses readchar 4.1.1.dev0. even though it says a few lines before readchar>=4.2.0 is installed

image

**Comment by ocefpaf** [comment]
It is very likely that the package metadata is wrong. See https://github.com/conda-forge/readchar-feedstock/pull/10/ for an attempt to fix it.

**Comment by nicornk** [comment]
@ocefpaf i think this is good to be merged. Not sure if it’s still maintained by @mxr-conda

File: recipe/meta.yaml [hunk 1]
```diff
@@ -41,8 +41,9 @@ about:
   license: MIT
   license_family: MIT
   license_file: LICENSE
-  summary: Collection of common interactive command line user interfaces, based on Inquirer.js
+  summary: Collection of common interactive command line user interfaces,  based on Inquirer.js
```
Comment (ocefpaf) [inline recipe/meta.yaml:44]:
```suggestion
  summary: Collection of common interactive command line user interfaces, based on Inquirer.js
```

**Pull request merged**